File: App_Clint/lib/features/wallet/domain/entities/bank_transfer_account.dart
import 'package:equatable/equatable.dart';

/// Active bank account for wallet top-up transfers (GET /api/v1/wallet/bank-transfer-account).
class BankTransferAccount extends Equatable {
  final int id;
  final String bankName;
  final String accountHolderName;
  final String accountNumber;
  final String? iban;
  final String? branchName;
  final String? instructions;
  final bool isActive;

  const BankTransferAccount({
    required this.id,
    required this.bankName,
    required this.accountHolderName,
    required this.accountNumber,
    this.iban,
    this.branchName,
    this.instructions,
    required this.isActive,
  });

  @override
  List<Object?> get props => [
        id,
        bankName,
        accountHolderName,
        accountNumber,
        iban,
        branchName,
        instructions,
        isActive,
      ];
}

File: Backend API/CleaningHouse_API/Models/Wallet/BankTransferAccount.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CleaningHouse_API.Models.Wallet;

[Table("BankTransferAccounts")]
public class BankTransferAccount
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string BankName { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    public string AccountHolderName { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    public string AccountNumber { get; set; } = string.Empty;

    [MaxLength(50)]
    public string? Iban { get; set; }

    [MaxLength(200)]
    public string? BranchName { get; set; }

    [MaxLength(1000)]
    public string? Instructions { get; set; }

    public bool IsActive { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
